splink/internals/realtime.py

```diff
@@ -0,0 +1,151 @@
+    else:
+        to_register_right = record_2
+
+    df_records_left = db_api.register_table(
+        to_register_left,
+        f"__splink__compare_records_left_{uid}",
+        overwrite=True,
+    )
+    df_records_left.templated_name = "__splink__compare_records_left"
+
+    df_records_right = db_api.register_table(
+        to_register_right,
+        f"__splink__compare_records_right_{uid}",
+        overwrite=True,
+    )
+    df_records_right.templated_name = "__splink__compare_records_right"
+
+    settings_id = id(settings)
+    if use_sql_from_cache:
+        if cached_sql := _sql_cache.get(settings_id, uid):
+            return db_api._sql_to_splink_dataframe(
+                cached_sql,
+                templated_name="__splink__realtime_compare_records",
+                physical_name=f"__splink__realtime_compare_records_{uid}",
+            )
+
+    if not isinstance(settings, SettingsCreator):
+        settings_creator = SettingsCreator.from_path_or_dict(settings)
+    else:
+        settings_creator = settings
+
+    settings_obj = settings_creator.get_settings(db_api.sql_dialect.sql_dialect_str)
+
+    retain_matching_columns = settings_obj._retain_matching_columns
+    retain_intermediate_calculation_columns = (
+        settings_obj._retain_intermediate_calculation_columns
+    )
+    settings_obj._retain_matching_columns = True
+    settings_obj._retain_intermediate_calculation_columns = True
+
+    pipeline = CTEPipeline([df_records_left, df_records_right])
+
+    cols_to_select = settings_obj._columns_to_select_for_blocking
+
+    select_expr = ", ".join(cols_to_select)
+    sql = f"""
+    select {select_expr}, 0 as match_key
+    from __splink__compare_records_left as l
+    cross join __splink__compare_records_right as r
+    """
+    pipeline.enqueue_sql(sql, "__splink__compare_two_records_blocked")
+
+    cols_to_select = settings_obj._columns_to_select_for_comparison_vector_values
+    select_expr = ", ".join(cols_to_select)
+    sql = f"""
+    select {select_expr}
+    from __splink__compare_two_records_blocked
+    """
+    pipeline.enqueue_sql(sql, "__splink__df_comparison_vectors")
+
+    sqls = predict_from_comparison_vectors_sqls_using_settings(
+        settings_obj,
+        sql_infinity_expression=db_api.sql_dialect.infinity_expression,
+    )
+    pipeline.enqueue_list_of_sqls(sqls)
+
+    if include_found_by_blocking_rules:
+        br_col = _select_found_by_blocking_rules(settings_obj)
+        sql = f"""
+        select *, {br_col}
+        from __splink__df_predict
+        """
+
+        pipeline.enqueue_sql(sql, "__splink__found_by_blocking_rules")
+
+    predictions = db_api.sql_pipeline_to_splink_dataframe(pipeline)
+    _sql_cache.set(settings_id, predictions.sql_used_to_create, uid)
+
+    settings_obj._retain_matching_columns = retain_matching_columns
+    settings_obj._retain_intermediate_calculation_columns = (
+        retain_intermediate_calculation_columns
+    )
```

Comment:
We probably don't need this in this case, as we only instantiate the `Settings` in this function